data-prepper-core/src/test/resources/logstash-filter.yaml

```diff
@@ -0,0 +1,14 @@
+logstash-converted-pipeline:
```

Comment:
Thanks for noticing this.

"data-prepper-core/src/test/resources/logstash-filter.yaml" should not have been added to the repo. It gets generated when the test is run. I'll remove this file.

"data-prepper-core/src/test/resources/logstash-filter.conf" is needed to test a significant part of the DataPrepperArgs constructor. I agree the conversion is not the goal of this test. We should refactor this part of the code to separate the file parsing logic. I moved the static function in DataPrepperArgs from another class to keep all related code in one place. The method itself is relative unchanged. I propose to refactor the static method into a configuration class that can handle the conversion logic outside of DataPrepperArgs in a future PR.

Reply:
It would be ideal if we could run the tests such that they generate files only in `build`. But if that is non-trivial, I'd say, let's add this file to `.gitignore` so that it doesn't get checked in again, or clutter our Git changes.

Reply:
For this PR we'll exclude the logstash-filter.yaml file with .gitignore and refactor the test so that the yaml file gets generated in a ./build directory in the future.